### PR TITLE
Encourage using the newtype pattern

### PR DESCRIPTION
**Pull request by gregwebs**
### What issue does this PR solve?

Encourage using the newtype pattern.

- to #26

### CODE REVIEW

File: contribute-to-tidb/code-style-and-quality-guide.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+
+## The newtype pattern improves code quality
+
+We can create a new type using the `type` keyword.
+
+The newtype pattern is perhaps most often used in Golang to get around type restrictions rather than to try to create new ones. It is used to create different interface implementations for a type or to extend a builtin type or a type from an existing package with new methods.
+
+However, it is generally useful to improve code clarity by marking that data has gone through either a validation or a transformation. Using a different type can reduce error handling and prevent improper usage. 
+
+```
```
Comment (tisonkun) [inline contribute-to-tidb/code-style-and-quality-guide.md:14]:
```suggestion
```go
```

**Review by tisonkun — APPROVED** [review]
Thanks for your contribution @gregwebs ! The content looks good to me.

Fix a render issue inline.

@zz-jason may you take a look or I will merge this pull request tomorrow.

**Comment by tisonkun** [comment]
@gregwebs I just notice that you told me that it is the first version. Shall we move it to merge now or we regard it as a draft and waiting for further inputs?

**Comment by gregwebs** [comment]
@tisonkun I would send separate PRs for anything else. This can be merged as is.

**Comment by tisonkun** [comment]
@gregwebs Thanks! Merging...

**Comment by tisonkun** [comment]
@all-contributors please add @gregwebs  for content

**Pull request merged**